src/rgw/rgw_s3select.cc

```diff
@@ -478,15 +478,18 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_parquet(const char* query)
   if (!m_s3_parquet_object.is_set()) {
     //parsing the SQL statement.
     s3select_syntax.parse_query(m_sql_query.c_str());
+    parquet_object::csv_definitions parquet;
 
   m_s3_parquet_object.set_external_system_functions(fp_s3select_continue,
 						fp_s3select_result_format,
 						fp_result_header_format,
 						fp_debug_mesg);
 
     try {
+      //setting the Parquet-reader properties. i.e. the buffer-size for the Parquet-reader
+      parquet::ceph::S3select_Config::getInstance().set_s3select_reader_properties(s->cct->_conf->rgw_parquet_buffer_size);
       //at this stage the Parquet-processing requires for the meta-data that reside on Parquet object 
-      m_s3_parquet_object.set_parquet_object(std::string("s3object"), &s3select_syntax, &m_rgw_api);
+      m_s3_parquet_object.set_parquet_object(std::string("s3object"), &s3select_syntax, &m_rgw_api, parquet);
```

Comment:
is this part of the fix of setting the memory size?

Reply:
no.
the s3select contains more changes not related to this PR.

this API interface change relates to output-format (JSON or CSV)
the format of statement result.

Reply:
the relevant change is https://github.com/ceph/s3select/pull/163

not always convenient to split changes in s3select-engine into separate RGW/s3select PR's.

Reply:
no need for a separate PR. just a separate commit.
this also makes backporting, rollbacks etc. much easier

Reply:
@galsalomon66 could you please point, which of the commits is the one that actually fix the memory issue?